AmplifyPlugins/Core/AWSPluginsCore/Auth/Provider/AmplifyAWSCredentialsProvider.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSCore
+
+public class AmplifyAWSCredentialsProvider: NSObject, AWSCredentialsProvider {
+
+    public func credentials() -> AWSTask<AWSCredentials> {
+        let completionSource = AWSTaskCompletionSource<AWSCredentials>()
+        _  = Amplify.Auth.fetchAuthSession { [weak self] event in
+
+            switch event {
+            case .completed(let session):
+                self?.parseAWSCredentialsFromSession(session, completionSource: completionSource)
+            case .failed(let error):
+                completionSource.set(error: error)
+            default:
+                let error = AuthError.unknown("Fetch auth session returned an invalid result type")
+                completionSource.set(error: error)
+            }
+        }
+        return completionSource.task
+    }
+
+    public func invalidateCachedTemporaryCredentials() {
+        guard let authPlugin = try? Amplify.Auth.getPlugin(for: "awsCognitoAuthPlugin")
+            as? AuthInvalidateCredentialBehavior else {
+                return
+        }
+
+        authPlugin.invalidateCachedTemporaryCredentials()
+    }
+
+    private func parseAWSCredentialsFromSession(_ session: AuthSession,
+                                                completionSource: AWSTaskCompletionSource<AWSCredentials>) {
+        let credentialsResult = (session as? AuthAWSCredentialsProvider)?.getAWSCredentials()
+        switch credentialsResult {
+        case .success(let credentials):
+            completionSource.set(result: credentials.toAWSCoreCredentials())
+        case .failure(let error):
+            completionSource.set(error: error)
+        case .none:
+            let error = AuthError.unknown("Auth session doesnot include aws credentials information")
```

Comment:
s/doesnot include aws/does not include AWS/

Reply:
https://github.com/aws-amplify/amplify-ios/pull/444/files